### PR TITLE
[ticket/15870] Add core.viewtopic_modify_forum_id

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-15870

Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15870

### CODE REVIEW

**Comment by 3D-I** [comment]
!set Event

File: phpBB/viewtopic.php [hunk 1]
```diff
@@ -263,6 +263,20 @@
 
 $forum_id = (int) $topic_data['forum_id'];
 
+/**
+ * Modify the forum ID to handle the correct display of viewtopic if needed
+ *
+ * @event core.viewtopic_modify_forum_id
+ * @var string	forum_id		forum ID
+ * @var array	topic_data		array of topic's data
+ * @since 3.2.4
```
Comment (marc1706) [inline phpBB/viewtopic.php]: We will not add any more events to the 3.2.4 release. Please change this to `3.2.5-RC1`, thanks.

Comment (3D-I) [reply phpBB/viewtopic.php]: Will do

**Comment by marc1706** [comment]
Rebased onto latest 3.2.x and squashed commits.

**Pull request merged**